divi/src/txmempool.cpp

```diff
@@ -23,6 +23,9 @@
 
 using namespace std;
 
+extern bool fAddressIndex;
+extern bool fSpentIndex;
+
```

Comment:
Might be preferable to inject these as const& dependencies  of the mempool class. The flags are already defined adjacent to the definition of the global mempool so it's not that much of a stretch. This also makes the dependency of the mempool on these a little more explicit in the instantiation.

Reply:
Sounds good, I'll update the PR accordingly.